### PR TITLE
Hometask1

### PR DESCRIPTION
**Pull request by accmi**

### CODE REVIEW

File: 2-building-blocks-of-oop-part-1/src/Point.ts [hunk 1]
```diff
@@ -1 +1,34 @@
-export class Point {}
+export class Point {
+  private x: number;
+  private y: number;
+
+  constructor();
```
Comment (GlebShaucou) [inline 2-building-blocks-of-oop-part-1/src/Point.ts:5]: constructor();
constructor(x: number, y: number);
constructor(x?: number, y?: number)

File: 2-building-blocks-of-oop-part-1/src/Shape.ts [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
 export abstract class Shape {
     abstract getType(): string;
+
+    protected color: string;
+    protected filled: boolean;
+
+    points: Point[];
+
+    constructor(points: Point[]);
+    constructor(points: Point[], color?: string, filled?: boolean) {
```
Comment (GlebShaucou) [inline 2-building-blocks-of-oop-part-1/src/Shape.ts:12]: constructor(points: Point[]);
constructor(points: Point[], color: string, filled: boolean)
constructor(points: Point[], color?: string, filled?: boolean)

File: 2-building-blocks-of-oop-part-1/src/Triangle.ts [hunk 3]
```diff
@@ -1,3 +1,60 @@
...
+  ISOSCELES_TRIANGLE = 'isosceles triangle',
+  SCALENE_TRIANGLE = 'scalene triangle',
+}
+
+export class Triangle extends Shape {
+  constructor(point1: Point, point2: Point, point3: Point);
+  constructor(point1: Point, point2: Point, point3: Point, color?: string, filled?: boolean) {
+    super([point1, point2, point3]);
+
+    /** if I run this I get this error: Expected 1 arguments, but got 3.
```
Comment (GlebShaucou) [inline 2-building-blocks-of-oop-part-1/src/Triangle.ts:15]: Try the following for superclass:

constructor(points: Point[]);
constructor(points: Point[], color: string, filled: boolean)
constructor(points: Point[], color?: string, filled?: boolean)

File: 2-building-blocks-of-oop-part-1/src/Shape.ts (hunk 2 above)
Comment (accmi) [reply 2-building-blocks-of-oop-part-1/src/Shape.ts:13]: done

File: 2-building-blocks-of-oop-part-1/src/Triangle.ts (hunk 3 above)
Comment (accmi) [reply 2-building-blocks-of-oop-part-1/src/Triangle.ts]: It worked, thank you

**Pull request merged**